### PR TITLE
Add `useDepositModal` hook

### PR DESCRIPTION
**Pull request by danieledamiani**

### CODE REVIEW

**Comment by ShawTim** [comment]
i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?

**Comment by danieledamiani** [comment]
> i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?

Yes, I've renamed it to `useDepositModalProps`

**Comment by ShawTim** [comment]
> > i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?
> 
> Yes, I've renamed it to `useDepositModalProps`

variable name has changed but the hook name is still `useDepositModal`

File: packages/tempus-client_v3/src/hooks/useDepositModal.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { maturityTerm$ } from './useMaturityTerm';
+
+export interface UseDepositModal {
+  precision: number;
+  poolStartDate: Date;
+  maturityTerms: MaturityTerm[];
+  usdRates: Map<Ticker, Decimal>;
+}
+
+export const [tempusPoolsForDepositModal$, setTempusPoolsForDepositModal] = createSignal<TempusPool[]>();
```
[the remark this answers is not in the document]
Comment (danieledamiani) [reply packages/tempus-client_v3/src/hooks/useDepositModal.ts:16]: I've tried different ways but none that worked well

**Comment by danieledamiani** [comment]
> > > i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?
> > 
> > 
> > Yes, I've renamed it to `useDepositModalProps`
> 
> variable name has changed but the hook name is still `useDepositModal`

Renamed the hook

File: packages/tempus-client_v3/src/hooks/useDepositModal.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShawTim) [reply packages/tempus-client_v3/src/hooks/useDepositModal.ts]: maybe accept `TempusPool[]` as params of `useDepositModalProp()`?

[the remark this answers is not in the document]
Comment (danieledamiani) [reply packages/tempus-client_v3/src/hooks/useDepositModal.ts]: it doesn't seem to work...

**Review by ShawTim — DISMISSED** [review]

File: packages/tempus-client_v3/src/config/ConfigManager.ts [hunk 2]
```diff
@@ -88,6 +114,41 @@ class ConfigManager {
...
+
+        return true;
+      })
+      .filter(pool => {
+        if (filterByProtocol) {
+          return pool.protocol === filterByProtocol;
+        }
+
+        return true;
+      })
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/config/ConfigManager.ts:142]: Can you replace this with `getFilteredPoolList`?

File: packages/tempus-client_v3/src/hooks/useDepositModalData.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  tokens: TokenMetadataProp;
+}
+
+export const [tempusPoolsForDepositModal$, setTempusPoolsForDepositModal] = createSignal<TempusPool[]>();
+
+const depositModalPools$ = tempusPoolsForDepositModal$.pipe(
+  filter(tempusPools => tempusPools.length > 0),
+  map(tempusPools => ({
+    tempusPools,
+    obsMaturityTerms: tempusPools.map(tempusPool => maturityTerm$(tempusPool)),
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/hooks/useDepositModalData.ts:21]: What does `obsMaturityTerms` mean?

Comment (danieledamiani) [reply packages/tempus-client_v3/src/hooks/useDepositModalData.ts:21]: `observable of MaturityTerms`

**Pull request merged**